src/charts/heatmap/heatmap.directive.js

```diff
@@ -246,14 +261,34 @@ angular.module('patternfly.charts').directive('pfHeatmap', function ($compile, $
       var redraw = function () {
         var data = scope.data;
         var color = d3.scale.threshold().domain(scope.thresholds).range(scope.heatmapColorPattern);
+        var rangeTooltip = d3.scale.threshold().domain(scope.thresholds).range(scope.rangeTooltips);
         var blocks;
         var fillSize = blockSize - scope.padding;
         var highlightBlock = function (block, active) {
           block.style('fill-opacity', active ? 1 : 0.4);
         };
+        var highlightBlockColor = function (block, fillColor) {
+          // Get fill color from given block
+          var blockColor = color(block.map(function (d) {
+            return d[0].__data__.value;
+          }));
+          // If given color matches, apply highlight
+          if (blockColor === fillColor) {
+            block.style('fill-opacity', 1);
+          }
+        };
+
+        // Highlight range on hover when default > fill size or when rangeOnHover is true
+        var highlightRange = ((scope.rangeOnHover === undefined && fillSize < 16) || scope.rangeOnHover) ? true : false;
+
         var svg = window.d3.select(thisComponent);
         svg.selectAll('*').remove();
 
+        // Highlight range on hover when default > fill size or when rangeOnHover is true
+        if ((scope.rangeOnHover === undefined && fillSize < 16) || scope.rangeOnHover) {
+          highlightRange = true;
+        }
+
```

Comment:
I meant when the fill size is less than the default value of 16.

If rangOnHover has not been set, the highlighting will be enabled when "when the fill size is less than the default value of 16". Otherwise rangOnHover=true/false overrides that.